File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/CreateProductTypeUseCase.java
package br.org.restaurantepopular.usecase.product_type;

import br.org.restaurantepopular.entity.ProductType;
import br.org.restaurantepopular.usecase.core.IUseCase;
import br.org.restaurantepopular.usecase.core.exception.ValidationException;

import java.util.Optional;

public class CreateProductTypeUseCase implements IUseCase<ProductType, Optional> {

    private final IProductTypeRepository productTypeRepository;

    public CreateProductTypeUseCase(IProductTypeRepository productTypeRepository) {
        this.productTypeRepository = productTypeRepository;
    }

    @Override
    public Optional execute(ProductType productType) {
        if (productType.getName() == null || productType.getName().isBlank() || productType.getName().isEmpty())
            throw new ValidationException("Product type name should not be empty");
            productTypeRepository.save(productType);
        return Optional.empty();
    }
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/CreateProductTypeUseCase.kt
package br.org.restaurantepopular.usecase.product_type

import br.org.restaurantepopular.entity.ProductType
import br.org.restaurantepopular.usecase.core.IUseCase
import br.org.restaurantepopular.usecase.core.exception.ValidationException

class CreateProductTypeUseCase(
    private val productTypeRepository: IProductTypeRepository
) : IUseCase<ProductType, ProductType> {

    override fun execute(request: ProductType): ProductType {
        if (request.name.isBlank())
            throw ValidationException("Product type name should not be empty")

        return productTypeRepository.save(request)
    }
}
